src/build/common/spriter.js

```diff
@@ -1,141 +1,192 @@
+  this.imageMap = merge(this.imageMap, imageMap);
+  this.sourceMap = merge(this.sourceMap, sourceMap);
+};
+
+exports.sprite = function (api, app, config, directories) {
+  var baseDirectory = config.outputResourcePath;
+  var relativeSpritesheetsDirectory = 'spritesheets';
+  var spritesheetsDirectory = path.join(baseDirectory,
+                                        relativeSpritesheetsDirectory);
+
+  return Promise.resolve(directories)
+    .map(function (directory) {
+      return exports.spriteDirectory(api, config, directory,
+                                     spritesheetsDirectory,
+                                     relativeSpritesheetsDirectory);
+    }, {concurrency: 1})
+    .reduce(function (allSheets, directorySheets) {
+      return allSheets.merge(directorySheets);
+    }, new SpriteSheetList())
+    .then(function (spritesheets) {
+      // create map of spritesheets to filenames
+      var filename = path.join(spritesheetsDirectory,
+                               'spritesheetSizeMap.json');
+
+      var sheetMap = {};
+      for (var i in spritesheets.imageMap) {
+        var img = spritesheets.imageMap[i];
+        if (!img || !img.sheet) {
+          continue;
+        }
 
-    var filteredPaths = [];
-
-    files.other = spriterOutput.other.map(function (filename) {
-      var fullPath = path.resolve(directories.src, filename);
-
-      if (path.basename(filename) === "metadata.json") {
-        try {
-          var filedata = fs.readFileSync(fullPath, "utf8");
-          var fileobj = JSON.parse(filedata);
-          if (fileobj.package === false) {
-            var filterPath = path.dirname(fullPath);
-            logger.log("Not packaging files from", filterPath);
-            filteredPaths.push(filterPath);
-          }
-        } catch (ex) {
-          logger.error("WARNING:", filename, "format is not valid JSON so cannot parse it.");
+        var sheet = img.sheet;
+        if (!sheetMap[sheet]) {
+          sheetMap[sheet] = {
+            w: img.sheetSize[0],
+            h: img.sheetSize[1]
+          };
         }
       }
 
-      return new File({
-        fullPath: fullPath,
-        target: path.join(directories.target, filename)
-      });
+      var sheetMapFile = new File({
+          base: baseDirectory,
+          path: filename,
+          contents: new Buffer(JSON.stringify(sheetMap))
+        });
+
+      sheetMapFile.inline = false;
+
+      return {
+        sourceMap: spritesheets.sourceMap,
+        files: [
+          // the image map maps each source image to the name of the
+          // spritesheet it was embedded in along with it's location and
+          // padding
+          new File({
+              base: baseDirectory,
+              path: path.join(spritesheetsDirectory, 'map.json'),
+              contents: new Buffer(JSON.stringify(spritesheets.imageMap))
+            }),
+          sheetMapFile
+        ]
+      };
+
+      /*
+      return fontsheetMap
```

Comment:
I suppose I can now.  It wasn't clear if native required the font sheet map file or not, but it's always empty regardless.